### PR TITLE
fix(cumulant-discovery): skip zero-gamma 0*inf on V(s') bootstrap

### PR DESCRIPTION
**Pull request by Kayvan-Zahiri**
## Summary
- Cumulant TD used `gamma * V(s')` while the constructor default is `gamma=0.0`; overflowed `V(s')` still forms IEEE `0*inf` NaN and freezes the surprise step.
- Skip the zero-gamma bootstrap with `_skip_zero_scale` before the product.
- Regression: `test_zero_gamma_does_not_multiply_overflow_v_next`.

## Test plan
- [x] `.venv/bin/python -m pytest tests/test_cumulant_discovery.py -q`
- [x] `.venv/bin/python -m ruff check alberta_framework/core/cumulant_discovery.py tests/test_cumulant_discovery.py`

Made with Cursor. No Slop measured-run receipt (not an approved Codex or Claude Code client).

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by kenjohnscreates — APPROVED** [review]
## Sound — the guard is live on the default path and covered

Approving. This is the cleanest of the four in the cluster. Recording what I verified so it doesn't need redoing.

**The guard is live, not defensive.** `gamma: float = 0.0` is the constructor default (`cumulant_discovery.py:197`) and the validator explicitly permits it (`lower=0.0`), so the zero branch is taken on every step under the default configuration — the opposite of a dead guard.

**No re-association.** The non-zero branch is literally `gamma * v_next`, and `cumulants + (...) - v` is untouched, so no parenthesis moves. Verified bitwise identical to main on finite inputs.

**Class A does not apply.** No `grad`/`vjp`/`jvp`/`linearize` reaches `cumulant_discovery.py`, so the `jnp.where` cannot kill a gradient here. Worth stating explicitly because it is the failure mode that made [#2593](https://github.com/elizaOS/asi/pull/2593) a regression.

**No asymmetry.** `gamma * v_next` is the only scaled term in `td`; `cumulants` and `-v` carry no multiplicative scale, so there is no unguarded complement.

**Covered, if narrowly.** Mutating the changed expression fails exactly 1 test out of 1455. Worth knowing which one: of the new test's four post-step assertions, only the age/step-advance assertion discriminates — the three `chex` finiteness assertions pass on unmutated main too, because main's NaN is caught by the hold condition rather than propagating. If you ever refactor that test, the age assertion is the one carrying the weight.

---

### Adjacent pre-existing bug, not introduced here — worth a separate look

While tracing gamma's provenance I found that the underflow guard protecting this exact parameter does not fire for the input type callers actually use. Flagging it here because it is the same parameter, not because it blocks this PR.

`_require_float32(..., preserve_nonzero=True)` is meant to reject a gamma that silently becomes zero in float32:

```python
if preserve_nonzero and numerator != 0 and stored == 0.0:
    raise ValueError(f"{name} must remain nonzero once narrowed to float32")
```

But `stored` is not the narrowed value. In `_float32_scalars.py:189`:

```python
stored = value if type(value) is float else narrowed
```

For a built-in `float` the original float64 is returned unnarrowed, so `stored == 0.0` is False for any tiny-but-nonzero literal and the check never fires. Measured on `main`:

```
value=1e-50   -> float32 = 0.0
  _require_float32("gamma", 1e-50, preserve_nonzero=True)            -> ACCEPTED  stored=1e-50
  _require_float32("gamma", np.float64(1e-50), preserve_nonzero=True) -> REJECTED  "gamma must remain nonzero once narrowed to float32"
```

Same value, opposite outcome, decided purely by Python type. Since `gamma: float = 0.0` is annotated `float`, callers take the bypassing branch essentially always. The consequence is quiet: `gamma=1e-45` gives you exactly zero bootstrapping while the validator that exists to catch precisely this says nothing.

`option_value_duration.py:98-99` implements the same intent correctly by re-narrowing first:

```python
narrowed = float(np.float32(stored))
if numerator != 0 and narrowed == 0.0:
    raise ValueError(f"{name} must not underflow to zero in float32")
```

Testing `narrowed` rather than `stored` in `cumulant_discovery.py:109` would close it. Happy to open a separate issue if that is preferred — it is orthogonal to this PR and I did not want to expand its scope.
